### PR TITLE
Lodash: Refactor away from `_.uniq()`

### PR DESCRIPTION
**Pull request by tyxla**
## What?
This PR removes the `_.uniq()` usage completely and deprecates the function. 

## Why?

Lodash is known to unnecessarily inflate the bundle size of packages, and in most cases, it can be replaced with native language functionality. See these for more information and rationale:

* https://github.com/WordPress/gutenberg/issues/16938#issuecomment-602837246
* https://github.com/WordPress/gutenberg/issues/17025
* https://github.com/WordPress/gutenberg/issues/39495 

## How?

`_.uniq( arr )` is easily replaced by native `[ ...new Set( arr ) ]`.

## Testing Instructions
* Verify building a release changelog still works: `npm run other:changelog`.
* Verify building docs still works without errors: `npm run docs:gen`.
* Testing for a few block types, verify the list of blocks to transform a block to is still the same, works well and contains unique block types.
* Insert a Post Date block inside a Post List block, uncheck "Default Format" and verify choosing a formst still works and contains unique date formats.
* Verify that when inserting tags for a post, the list is always unique and you can't add a duplicate.
* RN: Verify color palette still displays unique colors.
* Verify all tests still pass.

### CODE REVIEW

File: packages/block-editor/src/components/block-switcher/index.js [hunk 1]
```diff
@@ -64,7 +64,8 @@ export const BlockSwitcherDropdownMenu = ( { clientIds, blocks } ) => {
 				_icon = blockInformation?.icon; // Take into account active block variations.
 			} else {
 				const isSelectionOfSameType =
-					uniq( blocks.map( ( { name } ) => name ) ).length === 1;
+					[ ...new Set( blocks.map( ( { name } ) => name ) ) ]
```
Comment (gziolo) [inline packages/block-editor/src/components/block-switcher/index.js]: You could use [`Set.size`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/size) here.

File: bin/plugin/commands/changelog.js [hunk 2]
```diff
@@ -194,13 +194,15 @@ const REWORD_TERMS = {
...
 function getTypesByLabels( labels ) {
-	return uniq(
-		labels
-			.filter( ( label ) =>
-				Object.keys( LABEL_TYPE_MAPPING ).includes( label )
-			)
-			.map( ( label ) => LABEL_TYPE_MAPPING[ label ] )
-	);
+	return [
+		...new Set(
```
Comment (gziolo) [inline bin/plugin/commands/changelog.js:198]: Out of curiosity, is there any difference between spreading the Set and using [`Array.from`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from)?

Comment (tyxla) [reply bin/plugin/commands/changelog.js:198]: Good question! I guess the answer has a few different angles:

* Performance - they're almost equal, spreading being a bit faster: https://jsben.ch/5lKjg
* Functionality - they work the same way when working with iterables (which `Set`s are).
* Transpilation - the end result will differ slightly, but that would matter if this code was transpiled (which it isn't).

Unless you have strong feelings, I'd prefer to use spread as it's way more common around the codebase.

File: packages/block-editor/src/components/block-switcher/index.js (hunk 1 above)
Comment (tyxla) [reply packages/block-editor/src/components/block-switcher/index.js]: Good call, updated in https://github.com/WordPress/gutenberg/pull/43330/commits/ad02c59e4f568b6713f02c34f1f0a494e62f20d1

File: bin/plugin/commands/changelog.js (hunk 2 above)
Comment (gziolo) [reply bin/plugin/commands/changelog.js:198]: Screenshot 2022-08-17 at 17 51 24

Safari, macOS, so I guess it depends but it's nearly the same.

I don't have any objections to using spread here. In general, both approaches discussed are less readable than `uniq`. It isn't immediately obvious that `Set` is used to remove duplicates.

**Review by gziolo — APPROVED** [review]

**Comment by gziolo** [comment]
@tyxla, you are doing a great job refactoring all packages to remove `lodash` 🦸🏻

File: bin/plugin/commands/changelog.js (hunk 2 above)
Comment (sgomes) [reply bin/plugin/commands/changelog.js:198]: > It isn't immediately obvious that `Set` is used to remove duplicates.

I don't know if I agree with that; `[ ...new Set() ]` is a pretty common idiom these days.

**Pull request merged**